src/IdeaStatiCa.IntermediateModel/Extensions/ISIntermediateExtension.cs

```diff
@@ -0,0 +1,298 @@
+				default: throw new InvalidOperationException($"Unknown type of intermediateItem {intermediateItem.GetType()}");
+			}
+		}
+
+		/// <summary>
+		/// Get Element Value
+		/// </summary>
+		/// <param name="intermediateItem"></param>
+		/// <param name="property"></param>
+		/// <returns></returns>
+		/// <exception cref="InvalidOperationException"></exception>
+		public static string GetElementValue(this ISIntermediate intermediateItem, string property)
+		{
+			switch (intermediateItem)
+			{
+				case SObject sObject:
+					return sObject.GetElementValue(property);
+				case SPrimitive sPrimitive:
+					return sPrimitive.GetElementValue(property);
+				case SAttribute sAttribute:
+					return sAttribute.GetElementValue(property);
+				default: throw new InvalidOperationException($"Unknown type of intermediateItem {intermediateItem.GetType()}");
+			}
+		}
+
+		/// <summary>
+		/// Take (Get and Remove) Element Property
+		/// </summary>
+		/// <param name="intermediateItem"></param>
+		/// <param name="property"></param>
+		/// <returns></returns>
+		/// <exception cref="InvalidOperationException"></exception>
+		public static ISIntermediate TakeElementProperty(this ISIntermediate intermediateItem, string property)
+		{
+			switch (intermediateItem)
+			{
+				case SObject sObject:
+					return sObject.TakeElementProperty(property);
+				case SList sList:
+					return sList.TakeElementProperty(property);
+				case SPrimitive sPrimitive:
+					return sPrimitive.TakeElementProperty(property);
+				case SAttribute sAttribute:
+					return sAttribute.TakeElementProperty(property);
+				default: throw new InvalidOperationException($"Unknown type of intermediateItem {intermediateItem.GetType()}");
+			}
+		}
+
+
+		/// <summary>
+		/// Add Element Property
+		/// </summary>
+		/// <param name="intermediateItem"></param>
+		/// <param name="property"></param>
+		/// <exception cref="InvalidOperationException"></exception>
+		public static void AddElementProperty(this ISIntermediate intermediateItem, ISIntermediate property)
+		{
+			switch (intermediateItem)
+			{
+				case SObject sObject:
+					sObject.AddElementProperty(property);
+					break;
+				case SList sList:
+					sList.AddElementProperty(property);
+					break;
+				case SPrimitive sPrimitive:
+					sPrimitive.AddElementProperty(property);
+					break;
+				case SAttribute sAttribute:
+					sAttribute.AddElementProperty(property);
+					break;
+				default: throw new InvalidOperationException($"Unsupported type of intermediateItem {intermediateItem.GetType()}");
+			}
+		}
+
+		/// <summary>
+		/// Try Add Element Property
+		/// </summary>
+		/// <param name="intermediateItem"></param>
+		/// <param name="property"></param>
+		/// <returns></returns>
+		/// <exception cref="InvalidOperationException"></exception>
+		public static bool TryAddElementProperty(this ISIntermediate intermediateItem, ISIntermediate property)
+		{
+			switch (intermediateItem)
+			{
+				case SObject sObject:
```

Comment:
There are many switches in this class. 
C# 8 switch construct might make the class smaller

Reply:
Also not all have default case

Reply:
edit